src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,103 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalk = this.getMaxTalkIndex(talks);
+        for (final Talk talk : talks.active()) {
+            final List<String> requests = talk.read().xpath("//request");
+            if (requests.isEmpty()) {
+                int index = 0;
+                final List<XML> logs = talk.read().nodes(ARCHIVE_LOG);
+                if (logs.isEmpty()) {
+                    index = maxTalk + 1;
+                } else {
+                    index = this.getMaxLogIndex(logs) + 1;
+                }
+                talk.modify(
+                    new Directives().xpath("//talk").add("request")
+                        .attr(INDEX, Integer.toString(index))
+                        .attr("id", this.createRequestId())
+                        .add("type").set(INDEX)
```

Comment:
OK. What if I have XML like this:

```
<talk later="false" name="talk" number="1" public="true">
   <wire>
      <href>#1</href>
   </wire>
   <archive>
      <log index="1" />
   </archive>
</talk>
```
I need to add a node with the `index` attribute. What will this node look like? You pointed out earlier that `<request index=...>` is wrong. So, what node should I create instead of `<request index=...>`?